matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/summary/RoomSummaryDataSource.kt

```diff
@@ -232,6 +233,20 @@ internal class RoomSummaryDataSource @Inject constructor(
         }
     }
 
+    private fun RoomSummary.getWithParents(): RoomSummary {
+        val parents = flattenParentIds.mapNotNull { parentId ->
+            getRoomSummary(parentId)?.let { parentSummary ->
+                SpaceParentInfo(
```

Comment:
Why not just return the RoomSummaries here instead of fake parentInfo with fake/wrong canonical & viaservers?

Reply:
Idk why I didn't do this to start with lmao

Reply:
👍